Example/Example/AttachmentDetailsViewController.swift

```diff
@@ -7,10 +7,11 @@ class AttachmentDetailsViewController: UITableViewController
     @IBOutlet var alignmentSegmentedControl: UISegmentedControl!
     @IBOutlet var sizeSegmentedControl: UISegmentedControl!
     @IBOutlet var sourceURLTextField: UITextField!
+    @IBOutlet var linkURLTextField: UITextField!
     @IBOutlet var altTextField: UITextField!
 
     var attachment: ImageAttachment?
-    var onUpdate: ((ImageAttachment.Alignment, ImageAttachment.Size, URL, String?) -> Void)?
+    var onUpdate: ((ImageAttachment.Alignment, ImageAttachment.Size, URL, URL?, String?) -> Void)?
```

Comment:
This closure is getting a little unwieldy, particularly for the urls and strings. I wonder if we should either add a comment that clarifies what each parameter does, or simply add names to the parameters: 

```
var onUpdate: ((_ alignment: ImageAttachment.Alignment, _ size: ImageAttachment.Size, _ imageURL: URL, _ linkURL: URL?, _ altText: String?) -> Void)?
```

Reply:
The rest of the code seems to favor making things self-explanatory, rather than adding comments... so I think adding names to the parameters is a good idea. Making that change.

Unfortunately that doesn't make the point where you're using the closure any more readable since it's still:
`onUpdate(alignment.toAttachmentAlignment(), size.toAttachmentSize(), url, linkURL, alt)`

(because "[Function types cannot have argument labels](https://forums.developer.apple.com/thread/60346)" ?  meh)

Reply:
Right! But I guess at least in the sample you game, the variables you’re passing in all have explanatory names :)

Reply:
👍